htmresearch/frameworks/nlp/classification_model.py

```diff
@@ -176,17 +176,18 @@ def inferDocument(self, document, returnDetailedResults=False,
 
     lastTokenIndex = len(tokenList) - 1
     categoryVotes = numpy.zeros(self.numLabels)
-
+    voteTotals = numpy.zeros(self.numLabels)
     for i, token in enumerate(tokenList):
       votes, _, _ = self.inferToken(token,
                                     reset=int(i == lastTokenIndex),
                                     returnDetailedResults=False,
                                     sortResults=False)
 
-      if votes.sum() > 0:
-        # Increment the most likely category, breaking ties in a random fashion
-        topCategory = self._sortArray(votes)[0]
-        categoryVotes[topCategory] += 1
+      voteTotals += votes
+
+    # Increment the most likely category, breaking ties in a random fashion
+    topCategory = self._sortArray(voteTotals)[0]
+    categoryVotes[topCategory] += 1
```

Comment:
This is strange. Are you sure you want to always add an extra vote to the topCategory?  What if there aren't any times?  It's not clear we would always want a random tie breaker either - it is ok to have ties in many cases.  I think the application can apply this logic if it wants to - not sure we want to formalize it in the model.